code/modules/client/preferences_savefile.dm

```diff
@@ -374,6 +374,15 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 
 	return TRUE
 
+/datum/preferences/proc/sanitize_be_special(list/input_be_special)
+	var/list/output = list()
+
+	for (var/role in input_be_special)
+		if (role in GLOB.special_roles)
+			output += role
+
+	return output.len == input_be_special.len ? input_be_special : output
```

Comment:
Can't you just return output?

Reply:
Yes, though there's always something nice about validators returning their inputs. It's not used anywhere here, I just like it as a general expectance. I can remove it if you want.

Reply:
That's fine, I don't mind it